data-collection/data-collection/View Controllers/Layer Contents View Controllers/LayerContentsViewController.swift

```diff
@@ -0,0 +1,528 @@
+            
+            // If we're showing the layerContent, add it to our legend array.
+            if (configuration.layersStyle == .visibleLayersAtScale && showAtScale) || configuration.layersStyle == .allLayers {
+                if let featureCollectionLayer = layerContent as? AGSFeatureCollectionLayer {
+                    // only show Feature Collection layer if the sublayer count is > 1
+                    // but always show the sublayers (the call to `updateLayerLegend`)
+                    if featureCollectionLayer.layers.count > 1 {
+                        let internalLegendInfos = legendInfos[layerContent.objectIdentifier()] ?? []
+                        let rowConfiguration = LayerContentsRowConfiguration(LayerContentsRowConfiguration.Kind.layer(featureCollectionLayer), configuration: configuration, legendInfos: internalLegendInfos)
+                        rowConfiguration.isVisibleAtScale = showAtScale
+                        rowConfigurations.append(rowConfiguration)
+                    }
+                } else {
+                    let internalLegendInfos = legendInfos[layerContent.objectIdentifier()] ?? []
+                    var rowConfiguration: LayerContentsRowConfiguration
+                    var kind: LayerContentsRowConfiguration.Kind
+                    if let layer = layerContent as? AGSLayer {
+                        kind = LayerContentsRowConfiguration.Kind.layer(layer)
+                    }
+                    else {
+                        kind = LayerContentsRowConfiguration.Kind.sublayer(layerContent)
+                    }
+                    rowConfiguration = LayerContentsRowConfiguration(kind,
+                                                                     configuration: configuration,
+                                                                     legendInfos: internalLegendInfos)
+
+                    rowConfiguration.isVisibleAtScale = showAtScale
+                    rowConfigurations.append(rowConfiguration)
+                }
+                updateLayerLegend(layerContent, parentShowAtScale: showAtScale)
+            }
+        }
+        
+        // Our dictionary of child:parents was set up in `loadSublayersOrLegendInfos`.
+        // Now we need to populate the `Content.parents` array for each content.
+        // Start by looping through all `content` items.
+        rowConfigurations.forEach { configuration in
+            // Get the array of parents for `content`.
+            let parentArray = parents[configuration.kind] ?? []
+            configuration.indentationLevel = parentArray.count
+            
+            // For each parent object, find the matching Content.
+            parentArray.forEach { parent in
+                rowConfigurations.forEach { potentialParentContent in
+                    // Search all contents to see if it's a match.
+                    if potentialParentContent.kind.matches(object: parent) {
```

Comment:
Since `Kind` conforms to `Equatable`, this check could just test for equality:

```suggestion
                    if potentialParentContent.kind == .sublayer(parent) {
```

And then `matches(object:)` wouldn't be needed.